cmake/tools.cmake

```diff
@@ -53,20 +53,23 @@ list (GET COMPILER_VERSION_LIST 0 COMPILER_VERSION_MAJOR)
 # Example values: `lld-10`, `gold`.
 option (LINKER_NAME "Linker name or full path")
 
-if (NOT LINKER_NAME)
-    if (COMPILER_GCC)
-        find_program (LLD_PATH NAMES "ld.lld")
-        find_program (GOLD_PATH NAMES "ld.gold")
-    elseif (COMPILER_CLANG)
-        # llvm lld is a generic driver.
-        # Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead
-        if (OS_LINUX)
-            find_program (LLD_PATH NAMES "ld.lld-${COMPILER_VERSION_MAJOR}" "ld.lld")
-        elseif (OS_DARWIN)
-            find_program (LLD_PATH NAMES "ld64.lld-${COMPILER_VERSION_MAJOR}" "ld64.lld")
+# s390x doesnt support lld
```

Comment:
Okay, so the heuristics (l. 58ff) that picks a linker if none was explicitly specified (l. 54) doesn't work on s390x. I think we should rather assert that the user provided a linker explicitly. What about something like this:

```cmake
option (LINKER_NAME "Linker name or full path")

if (ARCH_S390X AND NOT LINKER_NAME)
    message (FATAL_ERROR "On s390x, the linker must be specified manually.")
endif ()

if (NOT LINKER_NAME)
    ...
```

Reply:
Since s390x doesnt support lld, it doesnt matter if linker_name is there or not. lld path should not be included in s390x case.

Reply:
For s390x the gnu linker will be used by default, so they should not have to supply the linker name. The preference for the LLVM linker (when it's installed) should be overridden in the s390x case.